lib/Dialect/HW/HWOpInterfaces.cpp

```diff
@@ -27,20 +27,24 @@ LogicalResult hw::verifyInnerSymAttr(InnerSymbolOpInterface op) {
   // If does not have any inner sym then ignore.
   if (!innerSym)
     return success();
-  if (isa<InstanceOp>(&op) || op->getNumResults() != 1) {
+
+  if (!op.supportsPerFieldSymbols()) {
     // The inner sym can only be specified on fieldID=0.
     if (innerSym.size() > 1 || !innerSym.getSymName()) {
-      op->emitOpError("cannot assign symbols to non-zero field id, for ops "
-                      "with zero or multiple results");
+      op->emitOpError("does not support per-field inner symbols");
       return failure();
     }
     return success();
   }
 
-  auto resultType =
-      op->getResultTypes()[0].dyn_cast<hw::FieldIDTypeInterface>();
+  auto result = op.getTargetResult();
+  // If op supports per-field symbols, but does not have a target result,
+  // its up to the operation to verify itself.
```

Comment:
Do you have a use case in mind ? No target result but still valid per-field inner sym, when can that occur ?

Reply:
No, just trying to be careful about assumptions.  Does this seem like a problem specifically, or more of a "this suggests that would be useful and as far as (we) can tell it's probably not so why pretend it's allowed?".  LMK re:former, and re:latter, good call.  I can add a comment saying this isn't currently intended to be how things work (and similarly on the interface method the other comment thread is on)?

Reply:
(Added comments as suggested here and at the other spot, thanks!!)